src/Compilers/VisualBasic/Portable/BoundTree/Statement.vb

```diff
@@ -569,7 +569,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                                 End If
 
                                 Dim stepOperand As IOperation = If(stepValue.IsConstant, DirectCast(stepValue, IOperation), New Temporary(SyntheticLocalKind.ForLoopStepValue, BoundFor, stepValue))
-                                statements.Add(New CompoundAssignment(controlReference, stepOperand, Semantics.Expression.DeriveAdditionKind(controlType), Nothing, stepValue.Syntax))
+                                statements.Add(New ExpressionStatement(controlReference, stepOperand, Semantics.Expression.DeriveAdditionKind(controlType), Nothing, stepValue.Syntax))
```

Comment:
I think biggest confusion comes from existence of this file (Expression/Statement - https://github.com/dotnet/roslyn/pull/18318#pullrequestreview-30331166)

those shouldn't be implementation of IOperation. I understand they are needed when Bound node directly implemented IOperation. now two are separated, code in expression.cs/statement.cs should be just factory rather than another IOperation implementation.